include/circt/Support/SymCache.h

```diff
@@ -0,0 +1,86 @@
+//===- SymCache.h - Declare Symbol Cache ------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares a Symbol Cache.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_SUPPORT_SYMCACHE_H
+#define CIRCT_SUPPORT_SYMCACHE_H
+
+#include "mlir/IR/SymbolTable.h"
+#include "llvm/Support/Casting.h"
+
+namespace circt {
+
+/// Base symbol cache class to allow for cache lookup through a pointer to some
+/// abstract cache. A symbol cache stores lookup tables to make manipulating and
+/// working with the IR more efficient.
+class SymbolCacheBase {
+public:
+  virtual ~SymbolCacheBase() {}
```

Comment:
Is there an appropriate library to sink this into?  https://llvm.org/docs/CodingStandards.html#provide-a-virtual-method-anchor-for-classes-in-headers